Failed to perform operation 'get'

Since upgrading to `com.google.cloud.bigtable:bigtable-hbase-1.x:1.0.0-pre1` I am occasionally seeing the following error. I've redacted some confidential elements of the stack trace. Any idea what may be causing it?

```
2017-07-12 18:35:06,260 [ERROR] XXX:67 Table xxx: Other exception thrown: Failed to perform operation. Operation='get', projectId='xxx', tableName='xxx', rowKey='xxx'
java.io.IOException: Failed to perform operation. Operation='get', projectId='xxx', tableName='xxx', rowKey='1406519542728802'
	at com.google.cloud.bigtable.hbase.BigtableTable.logAndCreateIOException(BigtableTable.java:515)
	at com.google.cloud.bigtable.hbase.BigtableTable.get(BigtableTable.java:266)
        ... frames omitted
Caused by: java.util.concurrent.CancellationException: Task was cancelled.
	at com.google.common.util.concurrent.AbstractFuture.cancellationExceptionWithCause(AbstractFuture.java:1123)
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:501)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:404)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:86)
	at com.google.cloud.bigtable.hbase.BigtableTable.get(BigtableTable.java:253)
	... 38 common frames omitted
```

@sduskis Here are the relevant portions of the stack traces with `1.0.0-pre2-SNAPSHOT`

```
2017-07-13 23:57:55,520 [ERROR] XXX:67 Table xxx: Other exception thrown: Failed to perform operation. Operation='get', projectId='xxx', tableName='xxx', rowKey='xxx'
java.io.IOException: Failed to perform operation. Operation='get', projectId='xxx', tableName='xxx', rowKey='xxx'
	at com.google.cloud.bigtable.hbase.BigtableTable.logAndCreateIOException(BigtableTable.java:528)
	at com.google.cloud.bigtable.hbase.BigtableTable.getResults(BigtableTable.java:269)
	at com.google.cloud.bigtable.hbase.BigtableTable.get(BigtableTable.java:255)
Caused by: java.util.concurrent.CancellationException: Task was cancelled.
	at com.google.common.util.concurrent.AbstractFuture.cancellationExceptionWithCause(AbstractFuture.java:1123)
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:501)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:380)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:86)
	at com.google.cloud.bigtable.hbase.BigtableTable.getResults(BigtableTable.java:264)
	... 39 common frames omitted
```

```
2017-07-13 23:57:55,519 [ERROR] c.g.c.b.h.BigtableTable:143 Encountered exception when executing get.
java.util.concurrent.CancellationException: Task was cancelled.
	at com.google.common.util.concurrent.AbstractFuture.cancellationExceptionWithCause(AbstractFuture.java:1123)
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:501)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:380)
	at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:86)
	at com.google.cloud.bigtable.hbase.BigtableTable.getResults(BigtableTable.java:264)
	at com.google.cloud.bigtable.hbase.BigtableTable.get(BigtableTable.java:255)
```

--- a/bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java
+++ b/bigtable-hbase-parent/bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestFilters.java
@@ -1185,6 +1185,18 @@ public class TestFilters extends AbstractTest {
     }
   }
 
+  @Test
+  public void testSingleValueFilterEmpty() throws IOException {
+    byte[] qual = dataHelper.randomData("testSingleValueFilterEmpty");
+    // Add {, }, and @ to make sure that they do not need to be encoded
+    byte[] rowKey = dataHelper.randomData("Empty");
+    byte[] empty = new byte[0];
+    try (Table table = getTable()) {
+      table.put(new Put(rowKey).addColumn(COLUMN_FAMILY, qual, empty));
+      assertKeysReturnedForSCVF(table, qual, CompareOp.EQUAL, empty, rowKey);
+    }
+  }
+
   private void assertKeysReturnedForSCVF(Table table, byte[] qualifier, CompareOp operator,
       byte[] value, byte[]... expectedKeys) throws IOException {
     SingleColumnValueFilter filter =

--- a/bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/ValueFilterAdapter.java
+++ b/bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/ValueFilterAdapter.java
@@ -75,8 +75,14 @@ public class ValueFilterAdapter extends TypedFilterAdapterBase<ValueFilter> {
       case LESS_OR_EQUAL:
         return createRowFilter(ValueRange.newBuilder().setEndValueClosed(value));
       case EQUAL:
-      return createRowFilter(
-        ValueRange.newBuilder().setStartValueClosed(value).setEndValueClosed(value));
+        if (comparator.getValue().length == 0) {
+          // The empty case does needs to use valueRegexFilter, since "end value closed" of empty
+          // is not allowed by the server.
+          return RowFilter.newBuilder().setValueRegexFilter(value).build();
+        } else {
+          return createRowFilter(
+            ValueRange.newBuilder().setStartValueClosed(value).setEndValueClosed(value));
+        }
       case NOT_EQUAL:
         // This strictly less than + strictly greater than:
         return RowFilter.newBuilder()

